ante/spamming_prevention.go

```diff
@@ -0,0 +1,118 @@
+package ante
+
+import (
+	"sync"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	oracletypes "github.com/umee-network/umee/x/oracle/types"
+)
+
+// SpammingPreventionDecorator will check if the transaction's gas is smaller than
+// configured hard cap
+type SpammingPreventionDecorator struct {
```

Comment:
nit: I would shorten the type name

```suggestion
type SpamPreventionDecorator struct {
```